test/unit/test_easy_logging.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+
+import os.path
+from pathlib import Path
+
+import pytest
+import tomlkit
+
+from snowflake.connector import EasyLoggingConfigPython
+from snowflake.connector.config_manager import CONFIG_MANAGER
+from snowflake.connector.constants import CONFIG_FILE
+
+THIS_DIR = os.path.dirname(os.path.realpath(__file__))
+TEMP_CONFIG_FILE_DIR = os.path.join(THIS_DIR, "..", "data")
+NONEXIST_PATH = os.path.join(THIS_DIR, "..", "data", "non_exist")
+INABSOLUTE_PATH = os.path.join("..", "data")
+INACCESSIBLE_PATH = os.path.join(THIS_DIR, "..", "data", "inaccessible")
+
+
+@pytest.fixture(scope="function")
+def config_file_setup(request):
+    param = request.param
+    # making different config file dir for each test to avoid race condition on modifying config.toml
+    os.makedirs(os.path.join(TEMP_CONFIG_FILE_DIR, param), exist_ok=True)
+    temp_config_file_path = os.path.join(TEMP_CONFIG_FILE_DIR, param, "config.toml")
+    CONFIG_MANAGER.file_path = Path(temp_config_file_path)
+
+    configs = {
+        "nonexist_path": {"log": {"save_logs": False, "path": NONEXIST_PATH}},
+        "inabsolute_path": {"log": {"save_logs": False, "path": INABSOLUTE_PATH}},
+        "inaccessible_path": {"log": {"save_logs": False, "path": INACCESSIBLE_PATH}},
+    }
+    # create inaccessible path and make it inaccessible
+    if not os.path.exists(INACCESSIBLE_PATH):
+        os.mkdir(INACCESSIBLE_PATH)
+    os.chmod(INACCESSIBLE_PATH, os.stat(INACCESSIBLE_PATH).st_mode & ~0o222)
+    try:
+        # create temp config file
+        with open(temp_config_file_path, "w") as f:
+            f.write(tomlkit.dumps(configs[param]))
+        yield
+    finally:
+        # remove created dir and file, including log paths and config file paths
+        CONFIG_MANAGER.file_path = CONFIG_FILE
+        os.remove(temp_config_file_path)
+        os.rmdir(os.path.join(TEMP_CONFIG_FILE_DIR, param))
+
+        if os.path.exists(NONEXIST_PATH):
+            os.rmdir(NONEXIST_PATH)
+        if os.path.exists(INACCESSIBLE_PATH):
+            os.rmdir(INACCESSIBLE_PATH)
+
+
+@pytest.mark.parametrize("config_file_setup", ["nonexist_path"], indirect=True)
+@pytest.mark.skipolddriver
+def test_config_file_nonexist_path(config_file_setup):
+    assert not os.path.exists(NONEXIST_PATH)
+    EasyLoggingConfigPython()
```

Comment:
If I'm reading the code right, doesn't this have to call `.create_log()` on the `EasyLoggingConfigPython` instance?  If not, then I think some other test does need to cover `.create_log()`.

Reply:
I planned to cover `.create_log()` in next step, which will initiate easy logging in snowflake connection and a integ test that test is as a whole

Reply:
By "next step" do you mean a follow up PR, or another commit in this PR?  A wise person once said "untested code is broken code!" 😄

Reply:
In the next follow up PR, I'll definitely add some test on that code

Reply:
I think two simple unit tests could be added here.  One that sets `save_log = True`, writes a log message, and verifies that the log message was written. You can use the [caplog](https://docs.pytest.org/en/latest/reference/reference.html#caplog) fixture.  The second would be similar but would set `save_log = False` and it would check that the log entry was _not_ written.

Reply:
I did some test, turns out caplog is not suitable here, since python connector will always produce log that caplog can capture. Instead, I am going to check whether there is log in `path` provided by config.toml. I think this is enough for testing `create_log()`